Lib/ufo2ft/featureWriters/kernFeatureWriter.py

```diff
@@ -147,242 +158,140 @@ def _write(self):
-            glyphs2=pair.side2,
+            glyphs2=cls._convertToFeaAst(pair.side2, ast_cache),
             valuerecord2=None,
             enumerated=enumerated,
         )
 
-    def _makeKerningLookup(
-        self, name, pairs, exclude=None, rtl=False, ignoreMarks=True
-    ):
-        assert pairs
-        rules = []
-        for pair in pairs:
-            if exclude is not None and exclude(pair):
-                self.log.debug("pair excluded from '%s' lookup: %r", name, pair)
-                continue
-            rules.append(
-                self._makePairPosRule(
-                    pair, rtl=rtl, quantization=self.options.quantization
-                )
-            )
-
-        if rules:
-            lookup = ast.LookupBlock(name)
-            if ignoreMarks and self.options.ignoreMarks:
-                lookup.statements.append(ast.makeLookupFlag("IgnoreMarks"))
-            lookup.statements.extend(rules)
-            return lookup
-
-    def _makeKerningLookups(self):
-        cmap = self.makeUnicodeToGlyphNameMapping()
-        if any(unicodeScriptDirection(uv) == "RTL" for uv in cmap):
-            # If there are any characters from globally RTL scripts in the
-            # cmap, we compile a temporary GSUB table to resolve substitutions
-            # and group glyphs by script horizontal direction and bidirectional
-            # type. We then mark each kerning pair with these properties when
-            # any of the glyphs involved in a pair intersects these groups.
-            gsub = self.compileGSUB()
-            dirGlyphs = classifyGlyphs(unicodeScriptDirection, cmap, gsub)
-            directions = self._intersectPairs("directions", dirGlyphs)
-            shouldSplit = "RTL" in directions
-            if shouldSplit:
-                bidiGlyphs = classifyGlyphs(unicodeBidiType, cmap, gsub)
-                self._intersectPairs("bidiTypes", bidiGlyphs)
+    @staticmethod
+    def _convertToFeaAst(
+        side: str | tuple[str, ...],
+        ast_cache: dict[str | tuple[str, ...], ast.GlyphName | ast.GlyphClass],
+    ) -> ast.GlyphName | ast.GlyphClass:
+        """Cache the conversion of a pair name or literal class to the Fea AST,
+        because we'll see the same literal classes over and over."""
+        if side in ast_cache:
+            return ast_cache[side]
+        if isinstance(side, str):
+            side_ast = ast.GlyphName(side)
+        else:
+            side_ast = ast.GlyphClass([ast.GlyphName(g) for g in side])
+        ast_cache[side] = side_ast
+        return side_ast
+
+    def _makeKerningLookup(self, name, ignoreMarks=True):
+        lookup = ast.LookupBlock(name)
+        if ignoreMarks and self.options.ignoreMarks:
+            lookup.statements.append(ast.makeLookupFlag("IgnoreMarks"))
+        return lookup
+
+    def knownScriptsPerCodepoint(self, uv):
+        if not self.context.knownScripts:
+            # If there are no languagesystems, consider everything common;
```

Comment:
> If there are no languagesystems

I think this comment is incomplete, in the sense that context.knownScripts doesn't only include the scripts defined in languagesystems but also those that are uniquely associated with cmapped characters that are kerned.